pytest/tests/loadtest/locust/common/linkdrop.py

```diff
@@ -0,0 +1,165 @@
+        super().__init__(account,
+                         contract.key.account_id,
+                         "create_drop",
+                         balance=int(1E23))
+        self.contract = contract
+        self.account = account
+        self.drop_id = drop_id
+
+    def args(self) -> dict:
+        return {
+            "public_keys": [],
+            "deposit_per_use": "100000000000000000000000",
+            "drop_id": self.drop_id
+        }
+
+    def sender_account(self) -> Account:
+        return self.account
+
+
+class AddKey(FunctionCall):
+
+    def __init__(self, linkdrop: Account, sender: Account, public_keys,
+                 drop_id: str):
+        super().__init__(sender,
+                         linkdrop.key.account_id,
+                         "add_keys",
+                         balance=int(15E22))
+        self.linkdrop = linkdrop
+        self.sender = sender
+        self.public_keys = public_keys
+        self.drop_id = drop_id
+
+    def args(self) -> dict:
+        return {"public_keys": self.public_keys, "drop_id": self.drop_id}
+
+    def sender_account(self) -> Account:
+        return self.sender
+
+
+class ClaimDrop(Transaction):
+
+    def __init__(
+        self,
+        sender: Account,
+        new_account_id: str,
+        la_public_key,  #this is the limited access key added to the linkdrop contract already
+        la_secret_key,  #same as above
+        node,
+        balance: int = 0,
+    ):
+        super().__init__()
+        self.sender = sender
+        self.receiver_id = sender.key.account_id
+        self.method = "create_account_and_claim"
+        self.new_account_id = new_account_id
+        # defensive cast to avoid serialization bugs when float balance is
+        # provided despite type hint
+        self.balance = int(balance)
+        self.node = node
+
+        #Create a new key pair, which would be a full access key to your account
+        private_key, public_key = ed25519.create_keypair()
+        # Convert keys to Base58 and add prefix
+        sk = 'ed25519:' + base58.b58encode(
+            private_key.to_bytes()).decode('ascii')
+        pk = 'ed25519:' + base58.b58encode(
+            public_key.to_bytes()).decode('ascii')
+        #need it for functions args
+        self.pk = pk
+
+        #Creating a signer with the limited access key
+        self.sender.key.sk = la_secret_key
+        self.sender.key.pk = la_public_key
+        self.la_public_key = la_public_key
+
+    def args(self) -> dict:
+        return {
+            "new_account_id": self.new_account_id,
+            "new_public_key": self.pk
+        }
+
+    def sign(self, block_hash) -> transaction.SignedTransaction:
+        return transaction.sign_function_call_transaction(
+            self.sender.key, self.receiver_id, self.method,
+            json.dumps(self.args()).encode('utf-8'), 100 * TGAS, self.balance,
+            self.sender.get_nonce_for_pk(self.node, self.sender.key.account_id,
```

Comment:
Is there any reason why you can't use `self.sender.use_nonce()` here instead of `get_nonce_for_pk`?
It is supposed to be a simpler and more scalable way because it caches the nonce betweek calls.

Reply:
The way the linkdrop contract works is that it exposes an add_key function which lets drop creator call it and add a new access key to the contract. 
The drop claimer then calls the contract function claim using this limited access key. So the contract is calling itself now basically. 

use_nonce gave you the nonce for the full key for the smart contract currently in the runner. I needed nonce for the newly added access key which is limited and can only call claim function.

Reply:
I see, In this case I suggest inlining the call to `mocknet_helpers.get_nonce_for_pk` here directly and removing the method `Account::get_nonce_for_pk(self, ...)`, as the `self` is not used in that method.